include/mfmg/dealii_operator.templates.hpp

```diff
@@ -0,0 +1,296 @@
+/*************************************************************************
+ * Copyright (c) 2017-2018 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_DEALII_OPERATOR_TEMPLATES_HPP
+#define MFMG_DEALII_OPERATOR_TEMPLATES_HPP
+
+#include <mfmg/dealii_operator.hpp>
+
+namespace mfmg
+{
+namespace dealii_adapter
+{
+
+template <typename VectorType>
+DealIIMatrixOperator<VectorType>::DealIIMatrixOperator(
+    std::shared_ptr<typename DealIIMatrixOperator<VectorType>::matrix_type>
+        matrix,
+    std::shared_ptr<dealii::SparsityPattern> sparsity_pattern)
+    : _sparsity_pattern(sparsity_pattern), _matrix(matrix)
+{
+  ASSERT(sparsity_pattern != nullptr,
+         "deal.II matrices require a sparsity pattern");
+
+  ASSERT(matrix != nullptr, "The matrix must exist");
+}
+
+template <typename VectorType>
+void DealIIMatrixOperator<VectorType>::apply(VectorType const &x,
+                                             VectorType &y) const
+{
+  _matrix->vmult(y, x);
+}
+
+template <typename VectorType>
+std::shared_ptr<Operator<VectorType>>
+DealIIMatrixOperator<VectorType>::transpose() const
+{
+  ASSERT_THROW_NOT_IMPLEMENTED();
+
+  return nullptr;
+}
+
+template <typename VectorType>
+std::shared_ptr<Operator<VectorType>>
+DealIIMatrixOperator<VectorType>::multiply(
+    Operator<VectorType> const &operator_b) const
+{
+  ASSERT_THROW_NOT_IMPLEMENTED();
+
+  return nullptr;
+}
+
+template <typename VectorType>
+std::shared_ptr<VectorType>
+DealIIMatrixOperator<VectorType>::build_domain_vector() const
+{
+  ASSERT_THROW_NOT_IMPLEMENTED();
```

Comment:
`build_domain_vector` and `build_range_vector` should probably be implemented for `DealIIMatrixOperator`. I guess they currently are not only because we don't use them.

Reply:
Yes, I think it's fine for now not having them